aten/src/ATen/ParallelCommon.cpp

```diff
@@ -103,6 +107,17 @@ int intraop_default_num_threads() {
   if (nthreads == 0) {
 #if defined(FBCODE_CAFFE2) && defined(__aarch64__)
     nthreads = 1;
+#elif defined(__aarch64__) && defined(__APPLE__)
+    // On Apple Silicon there are efficient and performance core
```

Comment:
This is true on most intel CPUs today as well right? Wouldn't this sysctl work for them as well on other unix flavors?

Reply:
Although `sysctl` API is present on both Linux and Mac, the particular key would only be available on MacOS